lib/protocol/context.ts

```diff
@@ -16,17 +18,19 @@ export const withCSM = () => {
   return process.env.INTEGRATION_WITH_CSM !== "off";
 };
 
-export const getProtocolContext = async (): Promise<ProtocolContext> => {
-  if (hre.network.name === "hardhat") {
-    const networkConfig = hre.config.networks[hre.network.name];
-    if (!networkConfig.forking?.enabled) {
-      await deployScratchProtocol(hre.network.name);
-    }
-  } else {
-    await deployUpgrade(hre.network.name);
+export const getProtocolContext = async (skipV3Contracts: boolean = false): Promise<ProtocolContext> => {
+  const isScratch = getMode() === "scratch";
+
+  if (isScratch) {
+    await deployScratchProtocol(hre.network.name);
+  } else if (process.env.UPGRADE) {
+    await deployUpgrade(hre.network.name, "upgrade/steps-upgrade-for-tests.json");
   }
 
-  const { contracts, signers } = await discover();
+  // TODO: revisit this when Pectra is live
```

Comment:
Looks like with the latest HardHat 2.24.0 we can remove this logic, since it's already Pectra